lib/src/main/java/io/ably/lib/realtime/ChannelBase.java

```diff
@@ -84,6 +85,12 @@ public abstract class ChannelBase extends EventEmitter<ChannelEvent, ChannelStat
 
     private int retryCount = 0;
 
+    /**
+     * This flag is determining whether the attach should be reattampted when a setConnected() event us receuved
+     * This is a workaround and the mechanism to handle this should be rethought in the future
+     * */
+    private final AtomicBoolean reattachOnConnectionResume = new AtomicBoolean(false);
```

Comment:
Can you explain what contention this is addressing?

Reply:
`setConnected()` and `reattach()` calls are done from different threads. setConnected() is called when a state transition happens - so it technically will always be done after `setConnected` , but as they arrive from different threads I wanted to minimize any sync issues by using an atomic boolean

Reply:
which threads? What triggers `reattach()` other than connection state changes?

My concern is that we're mixing synchronisation primitives; most operations involving connection state are performed by the `ConnectionManager` thread, and there is a lock there that protects the `ConnectionManager` state and ensures that all state transitions occur strictly serially.

Reply:
Here is a quick run to check in which thread this value is set and read.
```
2023-01-30 11:11:23.844  8588-8629  System.out              io.ably.lib.test                     I  thread_test reattachOnConnectionResume is set :WebSocketConnectReadThread-341
2023-01-30 11:11:23.845  8588-8617  System.out              io.ably.lib.test                     I  thread_test reattachOnConnectionResume is read :Thread-2
```
I think I misrenamed `reattach` as it isn't technically a reattach and will just set this flag so that this could be used when a state transition happens. I'm equally happy to make this non-atomic as I think operations still be ordered as you mentioned above.

Reply:
As discussed this is no longer the case as this information will be carried out with `StateIndication` c63183dbf8cf68ad1d5d0b60508205779f2defba